lib/site_prism/rspec_matchers.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module SitePrism
+  class RspecMatchers
+    def self.create_rspec_existence_matchers(element_name)
+      new(element_name).create_rspec_existence_matchers
+    end
+
+    def initialize(element_name)
+      @element_name = element_name
+    end
+
+    def create_rspec_existence_matchers
+      matcher = "have_#{element_name}"
```

Comment:
if you move all these to private methods you save more lines. I actually like what you've done here I think it looks good! It also would then tidy up the code-smell of items being passed 3 calls down when they're only needed at the bottom

Reply:
I'd opted not to to avoid having the method arguments shadow methods on the class but it's a small class so it's probably not a huge problem.

I don't think it's possible to avoid that codesmell without disabling rubocop method length limit. We need to have all those parameters in the local method scope at the time of defining the rspec matcher, otherwise we see `NameError`s when we actually try and invoke the matcher since it doesn't have access to the methods on this class at that point.

Abusing the method params like this was the only way I could come up with to get the required stuff in scope and still have enough lines available to actually define the matcher